src/mlpack/core/optimizers/fw/func_sq.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file func_sq.hpp
+ * @author Chenzhe Diao
+ *
+ * Square loss function: \f$ x-> 0.5 * || Ax - b ||_2^2 \f$.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_FW_FUNC_SQ_HPP
+#define MLPACK_CORE_OPTIMIZERS_FW_FUNC_SQ_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Square loss function \f$ f(x) = 0.5 * ||Ax - b||_2^2 \f$.
+ *
+ * Contains matrix \f$ A \f$ and vector \f$ b \f$.
+ */
+class FuncSq
+{
+ public:
+  /**
+   * Construct the square loss function.
+   *
+   * @param A matrix A.
+   * @param b vector b.
+   */
+  FuncSq(const arma::mat A, const arma::mat b) : A(A), b(b)
```

Comment:
Do you mean `const arma::mat& A, const arma::mat& b)` here?  Also if b should be a vector, maybe better to use `const arma::vec&`?